src/main/java/uk/gov/hmcts/reform/idam/health/backup/FileFreshnessProbe.java

```diff
@@ -0,0 +1,89 @@
+package uk.gov.hmcts.reform.idam.health.backup;
+
+import com.google.common.annotations.VisibleForTesting;
+import lombok.extern.slf4j.Slf4j;
+import uk.gov.hmcts.reform.idam.health.probe.HealthProbe;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.Clock;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.temporal.ChronoUnit;
+
+@Slf4j
+public class FileFreshnessProbe implements HealthProbe {
+
+    private final String probeName;
+    private final Path checkPath;
+    private final long fileExpiryMs;
+
+    private FileSystemInfo fileSystemInfo;
+    private Clock clock;
+
+    public FileFreshnessProbe(String probeName, String path, long fileExpiryMs) {
+        this.probeName = probeName;
+        this.checkPath = Paths.get(path);
+        this.fileExpiryMs = fileExpiryMs;
+        this.fileSystemInfo = new DefaultFileSystemInfo();
```

Comment:
Might consider instantiating this object out of the constructor and just passing it in as a constructor parameter. This would decouple the two classes + allow us to remove the setter method that's only used for testing.